app/lib/authentication/user.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# This User object represents an authenticated user
+# passed into this Application from Juntos Single Sign
+# On server.  The class wraps around a JWT (JSON Web Token)
+# that is generated on the SSO server.  This class
+# implements a convenient way to deserialize the JWT and
+# read important properties of the User.
+#
+# !This class is READ-ONLY! It reads a JWT, but cannot
+# change an User properties.
+class Authentication::User
+  # Note that you cannot instantiate this class directly. Instead,
+  # use the factory method `User.build`
+  private_class_method :new
+
+  ## Construction ############################################################
+  # Setting up a new user from a JWT user token
+
+  def initialize(user_token:)
+    @user_token                 = user_token
+    @user_payload, @jwt_header  = JWT.decode(@user_token, rsa_public_key, true, { algorithm: 'RS256' })
+
+    @user_payload = @user_payload.with_indifferent_access
+    @jwt_header   = @jwt_header.with_indifferent_access
+  end
+
+  # This factory method must be called to instantiate a new User object.
+  # Using this factory allows us to conditionally handle token decoding
+  # and verification issues
+  def self.build(user_token:, raise_errors:)
+    raise JWT::DecodeError, 'No user token' if user_token.blank?
+    new(user_token: user_token)
+
+  rescue  JWT::DecodeError, JWT::ExpiredSignature, JWT::InvalidIssuerError,
```

Comment:
There's a convenient place in the app to group errors and use it as 
```suggestion
rescue JWT_ERRORS
```
Checkout `config/initializers/errors.rb`

Reply:
nice!